File: 03.Singleton/java/src/singleton/singleton.java
package singleton;
public class singleton {
	private static singleton instance = null;
	public void work(){
		System.out.println("I am an instance and I am doing some work!");
	}
	public static singleton getInstance(){
		if (instance == null){
			instance = new singleton();
		}
		return instance;
	}
}

File: 03.Singleton/cpp/src/singleton.cpp
#include "singleton.h"
#include <cstddef>
#include <iostream>

singleton* singleton::instance = NULL;
singleton* singleton::getInstance(){
   if (!instance) {
       instance = new singleton;
   }
   std::cout << "Get instance called!" << std::endl;
   return instance;
}

singleton::singleton() {
    counter++;
    std::cout << "I have been instantiated only " << counter << " time." << std::endl;
}

singleton::singleton(const singleton& orig) { }
singleton::~singleton() { }

